ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/StrictMVar.hs

```diff
@@ -148,10 +152,22 @@ isEmptyMVar StrictMVar { tmvar } = atomically $ Lazy.isEmptyTMVar tmvar
 
 updateMVar :: MonadSTM m => StrictMVar m a -> (a -> (a, b)) -> m b
 updateMVar StrictMVar { tmvar, tvar, invariant } f = do
-    (a', b) <- atomically $ do
+    (!a', b) <- atomically $ do
         a <- Lazy.takeTMVar tmvar
         let (!a', b) = f a
```

Comment:
So we should put a bang on the outside of this tuple too?

Reply:
...instead of the current change.

Reply:
I don't think so. It's a pattern match in a lambda, so not a lazy pattern.